contracts/CliptoExchange.sol

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: AGPL-3.0 License
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./beacon/CloneableBeaconProxy.sol";
+import "./interfaces/ICliptoToken.sol";
+import "./CliptoExchangeStorage.sol";
+
+contract CliptoExchange is CliptoExchangeStorage, Initializable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+    uint256 private _feeNumer;
+    uint256 private _feeDenom;
```

Comment:
If this is going to be upgradable, state variables within the contract will cause major headaches later down the line.